src/libs/actions/Session/index.js

```diff
@@ -90,11 +90,29 @@ function signOutAndRedirectToSignIn() {
  * @param {String} [login]
  */
 function resendValidationLink(login = credentials.login) {
-    Onyx.merge(ONYXKEYS.ACCOUNT, {isLoading: true});
-    DeprecatedAPI.ResendValidateCode({email: login})
-        .finally(() => {
-            Onyx.merge(ONYXKEYS.ACCOUNT, {isLoading: false});
-        });
+    const optimisticData = [{
+        onyxMethod: CONST.ONYX.METHOD.MERGE,
+        key: ONYXKEYS.ACCOUNT,
+        value: {isLoading: true},
+    }];
+    const successData = [{
+        onyxMethod: CONST.ONYX.METHOD.MERGE,
+        key: ONYXKEYS.ACCOUNT,
+        value: {
+            isLoading: false,
+            message: Localize.translateLocal('resendValidationForm.linkHasBeenResent'),
+        },
+    }];
+    const failureData = [{
+        onyxMethod: CONST.ONYX.METHOD.MERGE,
+        key: ONYXKEYS.ACCOUNT,
+        value: {
+            isLoading: false,
+            message: '',
+        },
+    }];
+
+    API.read('RequestAccountValidationLink', {email: login}, {optimisticData, successData, failureData});
```

Comment:
this should be a `write`

Reply:
Gotcha, reason I made it a read is because `ResendValidateCode` is defined as a `read` command [here](https://github.com/Expensify/Web-Expensify/blob/75a1a9a99547e45327641436c2df8ae3e961ed4a/apiCommandTypes.php#L309-L309). I'll change it to `write` 👍 

We should probably define `RequestAccountValidationLink` as a `write` command in that same file in your Web-Expensify PR then?

@bondydaa

Reply:
updating here https://github.com/Expensify/Web-Expensify/pull/34670